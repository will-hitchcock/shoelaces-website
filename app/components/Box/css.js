import colors from 'utils/colors'

export const css = ({ color }) => ({
  padding: '1em',
  backgroundColor: colors[color || 'primary']
})

export default css
