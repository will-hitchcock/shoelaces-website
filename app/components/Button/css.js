import colors from 'utils/colors'

export const css = {
  display: 'inline-block',
  borderRadius: '3px',
  padding: '0.5rem 0',
  margin: '0.5rem 1rem',
  width: '11rem',
  background: colors.secondary,
  color: 'white',
  border: `2px solid ${colors.secondaryLighter}`,
  textAlign: 'center',
  textDecoration: 'none'
}

export default css
