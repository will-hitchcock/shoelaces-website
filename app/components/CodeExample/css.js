import colors from 'utils/colors'

const editor = {
  '.fontLoaded &': {
    fontFamily: '"Roboto Mono", monospace'
  },
  overflow: 'scroll'
}

const error = {
  borderLeft: `5px ${colors.darkRed} solid`,
  backgroundColor: colors.red,
  padding: '0.5em',
  color: colors.darkRed,
  marginTop: '0.5em'
}

export { editor, error }
