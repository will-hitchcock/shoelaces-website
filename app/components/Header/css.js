import colors from 'utils/colors'

const phi = (1 + Math.sqrt(5)) / 2
const shared = {
  textAlign: 'center',
  color: colors.primaryDark,
  margin: '0.25em 0'
}

const h1 = {
  fontSize: `${phi ** 3}em`,
  ...shared
}
const h2 = {
  fontSize: `${phi ** 2}em`,
  ...shared
}
const h3 = {
  fontSize: `${phi ** 1}em`,
  ...shared
}
const h4 = {
  fontSize: '1em',
  ...shared
}
const h5 = {
  fontSize: '1em',
  ...shared
}

export default { h1, h2, h3, h4, h5 }
