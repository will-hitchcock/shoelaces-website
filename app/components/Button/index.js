/**
*
* Button
*
*/
import React from 'react'
import { node, bool, string } from 'prop-types'
import styled from 'styled-components'
import { Link } from 'react-router'
import css from './css'

const StyledLink = styled(Link)`${css}`
const A = styled.a`${css}`

const Button = ({ children, external, path }) =>
  external ? (
    <A href={path} target="_blank">{children}</A>
  ) : (
    <StyledLink to={path}>{children}</StyledLink>
  )

Button.propTypes = {
  children: node,
  external: bool,
  path: string.isRequired
}

export default Button
