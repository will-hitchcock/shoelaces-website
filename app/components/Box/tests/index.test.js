import React from 'react'
import Button from '../index'
import css from '../css'

describe('Box', () => {
  it('renders a <Box />', () => {
    expect(shallow(<Button path="some/path" />)).toMatchSnapshot()
  })

  it('renders css', () => {
    expect(css).toMatchSnapshot()
  })
})
