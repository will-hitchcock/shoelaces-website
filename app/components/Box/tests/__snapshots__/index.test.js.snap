exports[`Box renders a <Box /> 1`] = `
<div
  className="sc-bdVaJa kKJIIQ" />
`;

exports[`Box renders css 1`] = `
Object {
  "backgroundColor": "#4682b4",
  "padding": "1em",
}
`;
